backend/src/shortcuts.rs

```diff
@@ -84,10 +89,11 @@ impl Shortcuts {
 
                         // How long is the shortest route through the original router, using this
                         // neighbourhood or not?
-                        let direct_length = match map
-                            .router_before
-                            .route_from_roads(*start_r, *end_r)
-                        {
+                        let direct_length = match map.router_before.route_from_roads(
```

Comment:
What's the significance of switching routers here?

Reply:
I'm actually not sure. I almost missed this at first and plugged `router_input_after`.

The directness is just for sorting and showing the "most likely" shortcuts first. If a shortcut through a neighbourhood is 100m long and the route using the entire map is 10m, then it's not a likely shortcut. The intention is just to see if using the main roads changes the distance a bunch. So maybe it's fine to use the full `map.router_after`? I think the reason that isn't done right now is performance -- we only call `rebuild_router` when absolutely necessary. Doing it every time you make any edit to a neighbourhood and need to recalculate shortcuts gets expensive.